sdk/src/oauth/state-machines/debug-oauth-2025-11-25.ts

```
@@ -1210,17 +1259,23 @@ export const createDebugOAuthStateMachine = (
 
-            // Validate PKCE support (REQUIRED for 2025-11-25)
+            // Validate PKCE support (REQUIRED for 2025-11-25).
+            //
+            // The MCP client requirement is to VERIFY S256 support before
+            // proceeding, so both shapes of failure — no advertised methods at
+            // all, and a list that omits S256 — have to stop the flow. Absent
+            // metadata throws here because there is nothing to show; a list
+            // without S256 is decided by `requiredMetadataEnforcement` below,
+            // so the debugger can still exercise a nonconforming server.
             const supportedMethods =
               authServerMetadata.code_challenge_methods_supported || [];
```

Biome (2.5.6)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)